src/SourceBuild/content/build.proj

```diff
@@ -43,6 +53,132 @@
+                            SourceName="local-feed"
+                            SourcePath="$(ScenarioTestsLocalFeedDir)" />
+  </Target>
+
+  <ItemGroup>
+    <_ArtifactsPackage Include="$(ArtifactsShippingPackagesDir)**;
+                                $(ArtifactsNonShippingPackagesDir)**"
+                       Condition="!$([System.String]::Copy('%(Identity)').EndsWith('.symbols.nupkg'))" />
+  </ItemGroup>
+
+  <Target Name="CreateScenarioTestsLocalFeed">
+
+    <RemoveDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <MakeDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <!-- Combine all packages into a single directory with symlinks so that we can have a single feed that points to them all. -->
+    <Copy SourceFiles="@(_ArtifactsPackage)"
+          DestinationFolder="$(ScenarioTestsLocalFeedDir)"
+          UseSymbolicLinksIfPossible="true" />
+
+  </Target>
+
+  <Target Name="CreateScenarioTestsDirectoryBuildFiles"
+          Outputs="@(_DirectoryProjFiles)">
+    <!-- Define Directory.Build.* files in the scenario tests artifacts directory to prevent the test run from
+         picking up the configuration from the VMR's Directory.Build.* files. We need an isolated configuration
+         from which to test. -->
+    <ItemGroup>
+      <_DirectoryProjFiles Include="$(ScenarioTestsArtifactsDir)Directory.Build.props" />
+      <_DirectoryProjFiles Include="$(ScenarioTestsArtifactsDir)Directory.Build.targets" />
+    </ItemGroup>
+    <WriteLinesToFile File="%(_DirectoryProjFiles.FullPath)"
+                      Lines="&lt;Project /&gt;"
+                      Overwrite="true" />
+  </Target>  
+
+  <Target Name="SetupScenarioTestsArtifacts"
+          DependsOnTargets="SetupNuGetConfig;
+                            CreateScenarioTestsLocalFeed;
+                            CreateScenarioTestsDirectoryBuildFiles">
+  </Target>
+
+  <PropertyGroup>
+    <ExtractDotNetSdkArchiveDependsOn>SetupScenarioTestsArtifacts;GetDotNetSdkArchive</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.zip'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkZip</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.tar.gz'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkTarball</ExtractDotNetSdkArchiveDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ExtractDotNetSdkArchive"
+          DependsOnTargets="$(ExtractDotNetSdkArchiveDependsOn)" />
+
+  <Target Name="ExtractDotNetSdkTarball">
+    <MakeDir Directories="$(DotNetInstallDir)" />
+    <Exec Command="tar -xzf @(SdkArchiveItem) -C $(DotNetInstallDir)" />
+  </Target>
+
+  <Target Name="ExtractDotNetSdkZip">
+    <MakeDir Directories="$(DotNetInstallDir)" />
+    <Unzip SourceFiles="@(SdkArchiveItem)" DestinationFolder="$(DotNetInstallDir)" />
+  </Target>
+
+  <Target Name="ExtractScenarioTestsPackage">
```

Comment:
nit
```suggestion
  <Target Name="ExtractScenarioTestsPackages">
```

Reply:
Fixed in https://github.com/dotnet/installer/pull/19222/commits/4ad40634d16a550ede3a66ebd4a058702e5facb7